### ISSUE TITLE
[Bug] [iOS Only] Expander: when first header is expanded, a second header selected expands incorrect header and content

### ISSUE THREAD

**Issue by jamiekeefer** [xamarin/XamarinCommunityToolkit#933]
### Description

When Item 1 is already expanded, selecting another expander header will not open the correct item in the CollectionView list.

### Steps to Reproduce

1. Create Expander with 5 expandable elements in a CollectionView
2. Expand first element
3. Select expandable element 2-5
4. Observe element 5 (or other) will expand.
5. Observe if element 2 (or any other than 1) is expanded, selecting another expandable item will not exhibit this issue.

### Expected Behavior
If header 1 is expanded and another header is selected is should show the correct header content.

### Actual Behavior
In my testing if item 1 was expanded, and I attempted to expand header 2, the content area would refresh and header 5 and its content would expand.

### Basic Information

Seems that having item 1 expanded is root of issue causing a different behavior when another header is selected. If any other item 2-5 is expanded, selecting another header would not force a reload of the content and simply expand the selected areas content.

- Version with issue: XCT 1.0.2
- Last known good version: N/A
- IDE: VS 2019 Enterprise
- Platform Target Frameworks: 
  - iOS: 14.4
  - Android: android behaves normally and without issue
  - UWP:  N/A
- Android Support Library Version: N/A
- Nuget Packages: N/A
- Affected Devices: iPhone 11iOS14.4 simulator

### Workaround
None

### Reproduction imagery

N/A

### Reproduction Link

N/A

**Comment by jamiekeefer** [comment]
Expander Header Markup

`<xct:Expander Tapped="OnExpanderTapped">
                                <xct:Expander.Header>
                                    <Grid>
                                        <StackLayout>
                                            <Label Text="Tote Id #:"/>
                                            <Label Text="{Binding Id}"
                                                FontAttributes="Bold"
                                                FontSize="Medium" />
                                        </StackLayout>
                                        <Image Source="expand.png"
                                            HorizontalOptions="End"
                                            VerticalOptions="Center">
                                            <Image.Triggers>
                                            <DataTrigger TargetType="Image"
                                                            Binding="{Binding Source={RelativeSource AncestorType={x:Type xct:Expander}}, Path=IsExpanded}"
                                                            Value="True">
                                                <Setter Property="Source"
                                                        Value="collapse.png" />
                                            </DataTrigger>
                                            </Image.Triggers>
                                        </Image>
                                    </Grid>
                                </xct:Expander.Header>`

**Comment by AndreiMisiukevich** [comment]
https://github.com/xamarin/XamarinCommunityToolkit/issues?q=is%3Aissue+is%3Aopen+expander

Can you please check if this is bug already reported?

**Comment by jamiekeefer** [comment]
@AndreiMisiukevich Looks like my issue is similar to this one here:  https://github.com/xamarin/XamarinCommunityToolkit/issues/608 albeit a little more specific.

Question: In your opinion, Is this more an issue with the toolkit or Mono Framework?

**Comment by jamiekeefer** [comment]
Suggested fix by @tommy-elliot-awh seems to be an interim solution: https://github.com/xamarin/XamarinCommunityToolkit/issues/608#issuecomment-759784901

> As stated in previous comments, Expander works fine with Scroll + Stack. Here I give more specifics about how to apply that as a workaround, when not requiring CollectionView or ListView specific functionality. Change the container from
> 
> <CollectionView ItemsSource="{Binding MyItems}">
>     <CollectionView.ItemTemplate>
>         
>     </CollectionView.ItemTemplate>
> </CollectionView>
> to
> 
> <ScrollView>
>     <StackLayout Spacing="0" BindableLayout.ItemsSource="{Binding MyItems}">
>         <BindableLayout.ItemTemplate>
>             
>         </BindableLayout.ItemTemplate>
>     </StackLayout>
> </ScrollView>
> This workaround resolves all expand/collapse issues on iOS and Android for me in this scenario (and also on Xamarin Forms 5.0), including the expand+scroll or collapse+scroll issue where when under CollectionView expand/collapse of one item would incorrectly expand/collapse other items further down/up the list once or twice each screen's worth of vertical space because of CollectionView cell reuse.

**Comment by jfversluis** [comment]
We've added to the [Docs](https://docs.microsoft.com/xamarin/community-toolkit/views/expander): Expander and CollectionView/ListView should not be used together. I'd love to have a solution for this but it’s one of those controls that don’t exist on the native platform. Because of that you will be fighting the OS on getting it right. Closing this one since there won't be any solution on the short-term unless someone contributes it.
